codepost-config.yaml

```diff
@@ -1,9 +1,3 @@
-course_name: COS999
-course_period: S2019
-api_key: 75015032bb226e6bb0f14aea0ff0d2a5d9a4b875 # Jeremie's API key
-tigerfile_path: /n/fs/tigerfile/Files/{course_name}_{course_period}/{assignment_name}
-tigerfile_path_space: "_"
-tests_path: "{pwd}/../.output/{submission}.output.txt"
-group_separator: "-"
-user_pattern: "{}@princeton.edu"
-partners_path: "{pwd}/partners.txt"
+course_name: <NAME HERE>
```

Comment:
I may be wrong, but I am not sure whether those will be parsed as string. It seems safer to have them be quoted, i.e., `"<NAME HERE>"` and so on. That way, if someone runs the tools without changing the configuration, the output will be "There is no course '<NAME HERE>'" rather than "Error parsing configuration file."

[On a side-note, will we have an "organization" structure? I expect the 'course_name' space to be rife with collisions.]

Reply:
Interesting. I'm not sure which error I like better. I think both are pretty informative:
* "Error parsing configuration file" is good because it will prompt users to open the configuration file and see what they need to add. It is bad because they will get an error when running supposedly "boilerplate" code, which could make them think our library is broken or they have downloaded it incorrectly. 
* "Your API key is invalid" will be the error produced if <___> is parsed as a string (not "There is no course named __"). Now that I write this, I think this error is worse, because it won't print the API key, so won't really explain to a user where to look to fix the error.

Reply:
Re: your side note, it's a good question. We are creating organizations (e.g. `princeton` vs `yale`) that allows us to create separate name spaces for each client institution, since a global course namespace would indeed be ripe with collisions as you point out. 

Formally, we are implementing a unique constraint for the following parameters of a `Course` object: `name`, `period`, `organization`. So, you can have `COS 101 | Spring` and `COS 101 | Fall` in the same organization, and you can also have `COS 101 | Spring` at Princeton and `COS 101 | Spring` at Yale without experiencing a collision.